src/netex_utils/mod.rs

```diff
@@ -32,6 +97,83 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
+
+    mod frame_type {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn parse_frame_type() {
+            let frame_type: FrameType = "ServiceFrame".parse().unwrap();
+            assert_eq!(frame_type, FrameType::Service);
+        }
+
+        #[test]
+        #[should_panic(expected = "Failed to convert \\'NotAFrameType\\' into a FrameType")]
+        fn parse_invalid_frame_type() {
+            "NotAFrameType".parse::<FrameType>().unwrap();
+        }
+    }
+
+    mod parse_frames_by_type {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn some_frame() {
+            let xml = r#"<root>
+                    <FareFrame />
+                    <ServiceFrame />
+                    <FareFrame />
+                </root>"#;
+            let root: Element = xml.parse().unwrap();
+            let frames = parse_frames_by_type(&root).unwrap();
+            assert_eq!(frames.keys().count(), 2);
+            assert_eq!(frames.get(&FrameType::Service).unwrap().len(), 1);
+            assert_eq!(frames.get(&FrameType::Fare).unwrap().len(), 2);
+        }
+
+        #[test]
+        #[should_panic(expected = "Failed to convert \\'UnknownFrame\\' into a FrameType")]
+        fn unknown_frame() {
+            let xml = r#"<root>
+                    <UnknownFrame />
+                </root>"#;
+            let root: Element = xml.parse().unwrap();
+            parse_frames_by_type(&root).unwrap();
+        }
+    }
+
+    mod get_only_frame {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn one_unit_price_frame() {
```

Comment:
The notion of "price" here is no longer valid in your tests (on get_only_frame)

Reply:
Indeed, I forgot to change the name of the method. Thanks. Done in e99ea23f84f10c886ec18969ed9ad85b7965a1b9.